### ISSUE TITLE
Language Server Snippets Definitions

### ISSUE THREAD

**Issue by johnrichardrinehart** [microsoft/vscode-cpptools#3354]
It seems that `vscode-cpptools` has recently transitioned from providing the language snippets in a JSON file to providing them via the language server (per #2945). However, I'm unable to find the definition of these snippets provided by the server to the client.

Can someone please point me to the definition of the snippets made available by the language server?

**Comment by sean-mcmanus** [comment]
The definitions are the same as https://github.com/Microsoft/vscode-cpptools/pull/2945/files#diff-4dc437ededb03985714ad3b401f14ec1 . We moved them in order to enable information from our lexer to programmatically enable/disable the snippets depending on the context.

We added the C_Cpp.suggestSnippets setting if you want to disable those. We could probably add some setting to some custom snippets file, but that isn't currently a high priority for us...if someone wants to create a pull request for that, we might be able to make the server side changes ourselves.
